streams/src/main/java/org/apache/kafka/streams/kstream/internals/KeyValuePrinter.java

```diff
@@ -69,11 +67,13 @@
         private Serde<?> keySerde;
         private Serde<?> valueSerde;
         private ProcessorContext processorContext;
+        private String streamName;
 
-        private KeyValuePrinterProcessor(PrintStream printStream, Serde<?> keySerde, Serde<?> valueSerde) {
+        private KeyValuePrinterProcessor(PrintStream printStream, Serde<?> keySerde, Serde<?> valueSerde, String streamName) {
             this.printStream = printStream;
             this.keySerde = keySerde;
             this.valueSerde = valueSerde;
+            this.streamName = (streamName == null) ? "" : "[" + streamName + "]: ";
```

Comment:
Why replace `null` by empty string `""` ? If user specifies `null` we can just print `null`, IMHO.